frontend/src/components/NavItem.tsx

```diff
@@ -0,0 +1,42 @@
+import { NavLink } from "react-router-dom";
+import navStyle from "./Nav.module.css";
+import classNames from "classnames";
+import { useEffect, useState } from "react";
+import { useCookies } from "react-cookie";
+
+export const NavItem = () => {
+  const [cookie] = useCookies(["token"]);
+  const [isCookieAvailable, setIsCookieAvailable] = useState(false);
+
+  useEffect(() => {
+    setIsCookieAvailable((prev) => !prev);
+  }, [cookie]);
+
+  return (
+    <li className={navStyle.li}>
+      {isCookieAvailable ? (
```

Comment:
L18-38 is more or less identical. Make a small custom component and concatenate.